lib/private/AllConfig.php

```diff
@@ -491,6 +491,45 @@ public function getUsersForUserValue($appName, $key, $value) {
 		return $userIDs;
 	}
 
+	/**
+	 * Gets the list of user ids sorted by lastLogin, from most recent to least recent
+	 *
+	 * @param int|null $limit how many users to fetch
+	 * @param int $offset from which offset to fetch
+	 * @param string $search search users based on search params
+	 * @return array of user IDs
+	 */
+	public function getLastLoggedInUsers(?int $limit = null, int $offset = 0, string $search = ''): array {
+		// TODO - FIXME
+		$this->fixDIInit();
+
+		$query = $this->connection->getQueryBuilder();
+
+		$query->select('uid', 'displayname')
+			->from('users', 'u')
+			->leftJoin('u', 'preferences', 'p', $query->expr()->andX(
+				$query->expr()->eq('userid', 'uid'),
+				$query->expr()->eq('appid', $query->expr()->literal('login')),
+				$query->expr()->eq('configkey', $query->expr()->literal('lastLogin')))
+			)
+			// sqlite doesn't like re-using a single named parameter here
+			->where($query->expr()->iLike('uid', $query->createPositionalParameter('%' . $this->connection->escapeLikeParameter($search) . '%')))
+			->orWhere($query->expr()->iLike('displayname', $query->createPositionalParameter('%' . $this->connection->escapeLikeParameter($search) . '%')))
+			->orWhere($query->expr()->iLike('configvalue', $query->createPositionalParameter('%' . $this->connection->escapeLikeParameter($search) . '%')))
+			->orderBy($query->func()->lower('configvalue'), 'DESC')
+			->addOrderBy('uid_lower', 'ASC')
+			->setFirstResult($offset)
+			->setMaxResults($limit);
+
+		$result = $query->executeQuery();
+		$displayNames = [];
```

Comment:
```suggestion
		$result = $query->executeQuery();
		$uids = $result->fetchAll(\PDO::FETCH_COLUMN);
		$result->closeCursor();
		return $uids;
```
Like that, if you only select uid with distinct you can use this simple code block instead of a while.